ProjFS.Mac/PrjFSKextLogDaemon/PrjFSKextLogDaemon.cpp

```diff
@@ -155,6 +165,120 @@ static os_log_type_t KextLogLevelAsOSLogType(KextLog_Level level)
     }
 }
 
+static void LogPanics()
+{
+    DIR* dr = opendir(PanicLogDirectory);
+    if (dr == nullptr)
+    {
+        ostringstream errorMessage;
+        errorMessage << "Unable to open " << PanicLogDirectory << ", errno=" << errno << ", errorstr=" << strerror(errno);
+        LogDaemonError(errorMessage.str());
+    }
+
+    // Look up the last run time
+    long lastRunTimestamp = GetFileModifiedTime(PanicLogTimestampFile);
+
+    struct dirent* dirEntry;
+    while ((dirEntry = readdir(dr)) != nullptr)
+    {
+        // Only check files
+        if(dirEntry->d_type == DT_REG)
+        {
+           // Check for .panic extension
+           const char* ext = strrchr(dirEntry->d_name,'.');
+           if(ext != nullptr && strcmp(ext, ".panic") == 0)
+           {
+               // Get the full name of the panic file
+               string fullFileName = PanicLogDirectory;
+               fullFileName += "/";
+               fullFileName += dirEntry->d_name;
+
+               // Verify it's after the last timestamp
+               long panicLogTime = GetFileModifiedTime(fullFileName.c_str());
+               if (panicLogTime == 0)
+               {
+                   // Record error if we can't determine timestamp
+                   ostringstream errorMessage;
+                   errorMessage << "Unable to run stat on panic at " << fullFileName.c_str() << "\n";
+                   const char* errorMessageString = errorMessage.str().c_str();
+                   LogKextMessage(OS_LOG_TYPE_DEFAULT, 0, errorMessageString, static_cast<int>(strlen(errorMessageString)));
+                   continue;
+               }
+
+               if (panicLogTime >= lastRunTimestamp)
+               {
+                  // Verify 'org.vfsforgit.PrjFSKext(' occurs in the log
+                  FILE* fp = fopen(fullFileName.c_str(), "r");
+                  if (fp)
```

Comment:
Log if the `fopen` fails?

Reply:
@jeschu1 what are you thoughts here, do you think we should log if `fopen` fails?

Reply:
Probably. It's extremely unlikely that it will fail, but if it does, it suggests the system is in a bad state (overzealous security software, corrupted file system, hardware issues, etc.) so any VFSForGit errors are to be taken with a grain of salt.

Reply:
Also, nit: `if (fp != nullptr)`

Reply:
I definitely agree @wilbaker @pmj, I'm adding a guard against fopen failing.